drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,717 @@
+uint8_t sx127x_get_op_mode(sx127x_t *dev)
+{
+    return sx127x_reg_read(dev, SX127X_REG_OPMODE) & ~SX127X_RF_OPMODE_MASK;
+}
+
+void sx127x_set_op_mode(sx127x_t *dev, uint8_t op_mode)
+{
+    static uint8_t op_mode_prev = 0;
+
+    op_mode_prev = sx127x_get_op_mode(dev);
+
+    if (op_mode != op_mode_prev) {
+        /* Replace previous mode value and setup new mode value */
+        sx127x_reg_write(dev, SX127X_REG_OPMODE,
+                         (op_mode_prev & SX127X_RF_OPMODE_MASK) | op_mode);
+    }
+}
+
+sx127x_lora_bandwidth_t sx127x_get_bandwidth(sx127x_t *dev)
+{
+    return dev->settings.lora.bandwidth;
+}
+
+inline void _low_datarate_optimize(sx127x_t *dev)
+{
+    switch (dev->driver_type) {
+    case SX1276_TYPE:
+        if ( ((dev->settings.lora.bandwidth == SX127X_BW_125_KHZ) &&
+              ((dev->settings.lora.datarate == SX127X_SF11) ||
+               (dev->settings.lora.datarate == SX127X_SF12))) ||
+             ((dev->settings.lora.bandwidth == SX127X_BW_250_KHZ) &&
+              (dev->settings.lora.datarate == SX127X_SF12))) {
+            dev->settings.lora.low_datarate_optimize = 0x01;
+        } else {
+            dev->settings.lora.low_datarate_optimize = 0x00;
+        }
+
+        sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG3,
+                         (sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG3) &
+                          SX127X_RF_LORA_MODEMCONFIG3_LOWDATARATEOPTIMIZE_MASK) |
+                         (dev->settings.lora.low_datarate_optimize << 3));
+        break;
+    case SX1272_TYPE:
+        if ( (dev->settings.lora.bandwidth == SX127X_BW_125_KHZ) &&
+             ((dev->settings.lora.datarate == SX127X_SF11) ||
+              (dev->settings.lora.datarate == SX127X_SF12))) {
+            dev->settings.lora.low_datarate_optimize = 0x01;
+        } else {
+            dev->settings.lora.low_datarate_optimize = 0x00;
+        }
+
+        sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1,
+                         (sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1) &
+                          SX127X_RF_LORA_MODEMCONFIG1_LOWDATARATEOPTIMIZE_MASK) |
+                         (dev->settings.lora.low_datarate_optimize));
+        break;
+    default:
+        DEBUG("Unknown device type :%d\n", dev->driver_type);
+        break;
+    }
+}
+
+inline void _update_bandwidth(sx127x_t *dev)
+{
+    uint8_t config1_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1);
+    switch(dev->driver_type) {
+    case SX1276_TYPE:
+        config1_reg &= SX1276_RF_LORA_MODEMCONFIG1_BW_MASK;
+        switch (dev->settings.lora.bandwidth) {
+        case SX127X_BW_125_KHZ:
+            config1_reg |= (SX1276_RF_LORA_MODEMCONFIG1_BW_125_KHZ << 4);
+            break;
+        case SX127X_BW_250_KHZ:
+            config1_reg |=  SX1276_RF_LORA_MODEMCONFIG1_BW_250_KHZ << 4;
+            break;
+        case SX127X_BW_500_KHZ:
+            config1_reg |=  SX1276_RF_LORA_MODEMCONFIG1_BW_500_KHZ << 4;
+            break;
```

Comment:
Either remote shifting or change SX1276_RF_LORA_MODEMCONFIG1_BW_XXX_KHZ for SX1276 device
For example, in the current state SX1276_RF_LORA_MODEMCONFIG1_BW_125_KHZ is 0x70. So 0x70 << 4 We lost our bits. 
SX1272 seems fine.